### DISCUSSION TITLE
Sketchware pro

### DISCUSSION THREAD

**Discussion by DevVigilante** [Sketchware-Pro/Sketchware-Pro#712] [General]
> Idea : There should be a wedget(view) labelled with "Made with Sketchware" it should be optional if a user wants to show that this app is made with Sketchware

> Idea : There should be a screen that enables user create his own library

> Feature request : Sketchware pro must a feature that enables a user to create listener for specific component

> Idea  : If Sketchware pro contain an option that enables a user to deploy his `Project` or his `Own Library (as Idea2)` to GitHub then all users will 100% like it.

Please reply or inform me Sketchware pro team if you are going to add it in upcoming version.

When will you release new version of Sketchware pro will Inform us🥰.

Thanks to Sketchware pro team

**Comment by PranavPurwar** [comment]
Except the feature request 3, none of the above features are going to be implemented. Third feature too may not get implemented due to custom components and custom blocks.

**Reply by DevVigilante** [reply]
It will implement?

**Reply by PranavPurwar** [reply]
there is low possibility but maybe

**Reply by PranavPurwar** [reply]
ask Jbk0 or Hasrat

**Reply by DevVigilante** [reply]
@JavkhlanK and @hasrat-ali ?

**Comment by j-px** [comment]
what exactly do you mean by "specific" Component? You already can specify what Component the Listener's Events are available for.

**Reply by DevVigilante** [reply]
> what exactly do you mean by "specific" Component? You already can specify what Component the Listener's Events are available for.

I mean that If I created a component then i need to create listener for them there is no feature to add it.Listener for components are only available for in-built component.

**Reply by j-px** [reply]
They're not. How else would you use Custom Listeners then?

**Reply by j-px** [reply]
I'm not 100 % sure how to add them (since I never worked with Custom Blocks, Components, Events, etc.) before, but it is possible.

**Reply by DevVigilante** [reply]
> I'm not 100 % sure how to add them (since I never worked with Custom Blocks, Components, Events, etc.) before, but it is possible.

If it is possible then can you give me a example component with listener?
Please So I can understand it logic.

**Reply by DevVigilante** [reply]
> I'm not 100 % sure how to add them (since I never worked with Custom Blocks, Components, Events, etc.) before, but it is possible.

It is possible for wedget/views but not for components.

**Reply by j-px** [reply]
I hope this is accurate.

1. First create a Custom Component, giving it a name, ID, type name, type class, and build class (and more if wanted)
2. Then create a new Listener, making its class/method independent is probably needed.
3. Add Events to that then. Make sure to use the same Component name as specified earlier.

And that's what I think is all needed!

**Reply by DevVigilante** [reply]
> I hope this is accurate.
> 
> 1. First create a Custom Component, giving it a name, ID, type name, type class, and build class (and more if wanted)
> 2. Then create a new Listener, making its class/method independent is probably needed.
> 3. Add Events to that then. Make sure to use the same Component name as specified earlier.
> 
> And that's what I think is all needed!

Its worked thanks a lot but I don't know how to add code placed in event by putting blocks
